modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/display_layout/edit_asset_category_cp_display_layout.jsp

```diff
@@ -99,163 +99,8 @@ if (cpDisplayLayout != null) {
-		link.setAttribute('href', '#' + id);
-		link.innerText = title;
-		linkContainer.appendChild(link);
-
-		var buttonContainer = document.createElement('span');
-		buttonContainer.classList.add('label-item');
-		buttonContainer.classList.add('label-item-after');
-
-		var button = document.createElement('button');
-		button.classList.add('close');
-		button.setAttribute('aria-label', 'Close');
-		button.setAttribute('type', 'button');
-		button.addEventListener('click', handleCloseButtonClick);
-
-		var svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
-		svg.classList.add('lexicon-icon');
-		svg.classList.add('lexicon-icon-times');
-		svg.setAttribute('focusable', false);
-		svg.setAttribute('role', 'presentation');
-
-		var use = document.createElementNS('http://www.w3.org/2000/svg', 'use');
-		use.setAttribute(
-			'href',
-			Liferay.ThemeDisplay.getPathThemeImages() + '/clay/icons.svg#times'
-		);
-
-		svg.appendChild(use);
-		button.appendChild(svg);
-		buttonContainer.appendChild(button);
-
-		labelContainer.appendChild(linkContainer);
-		labelContainer.appendChild(buttonContainer);
-
-		return labelContainer;
-	}
-
-	function handleCloseButtonClick(event) {
-		var button = event.currentTarget;
-		categoriesContainer.removeChild(button.parentElement.parentElement);
-	}
-
-	if (assetCategoryId) {
-		var categoryNode = createLabel(
-			'category-' + assetCategoryId,
-			assetCategoryName
-		);
-		categoriesContainer.appendChild(categoryNode);
-		classPK.value = assetCategoryId;
-	}
-
-	window.document
-		.querySelector('#<portlet:namespace />selectCategories')
-		.addEventListener('click', (event) => {
-			Liferay.Util.openSelectionModal({
-				buttonAddLabel: '<liferay-ui:message key="done" />',
-				multiple: true,
-				onSelect: (selectedItems) => {
-					if (selectedItems) {
-						Object.keys(selectedItems).forEach((key) => {
-							var item = selectedItems[key];
-
-							if (!item.unchecked && categoriesContainer) {
-								var categoryNodes = categoriesContainer.children;
-
-								if (categoryNodes.length) {
-									for (var i = 0; i < categoryNodes.length; i++) {
-										categoriesContainer.removeChild(
-											categoryNodes.item(i)
-										);
-									}
-								}
-
-								delete selectedItems.key;
-
-								var categoryNode = createLabel(
-									'category-' + item.categoryId,
-									item.value
-								);
-
-								categoriesContainer.appendChild(categoryNode);
-								classPK.value = item.categoryId;
-
-								return;
-							}
-						});
-					}
-				},
-				selectEventName: '<portlet:namespace />selectCategory',
-				title: '<liferay-ui:message key="select-category-display-page" />',
-				url:
-					'<%= categoryCPDisplayLayoutDisplayContext.getCategorySelectorURL(renderResponse) %>',
-			});
-		});
-</aui:script>
\ No newline at end of file
+<liferay-frontend:component
+	componentId="EditAssetCategoryCPDisplayLayout"
+	context="<%= categoryCPDisplayLayoutDisplayContext %>"
```

Comment:
@kresimir-coko  This seems wrong, not all contexts are the same. It depends on the context. I know, mind === blown.

See how we are doing this in https://github.com/liferay-commerce/liferay-portal/pull/1057/commits/501d7a4f7469dcb90036a9fa01ff60015bb55979

Reply:
Thanks @markocikos !